robot_sf/gym_env/robot_env_from_base.py

```
@@ -0,0 +1,249 @@
+RobotPose = Tuple[Vec2D, float]
+
+
+class RobotEnvFromBase(BaseEnv):
+    """
+    Representing a Gymnasium environment for training a self-driving robot
+    with reinforcement learning.
+    """
+
+    def __init__(
+        self,
+        env_config: EnvSettings = EnvSettings(),
```

Ruff (0.8.2)

Do not perform function call `EnvSettings` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)